src/Uno.UI/UI/Xaml/FrameworkElement.Layout.wasm.cs

```diff
@@ -209,11 +209,23 @@ private void ArrangeNative(Point offset, Size oldRenderSize)
 					throw new InvalidOperationException($"{this}: Invalid frame size {newRect}. No dimension should be NaN or negative value.");
 				}
 
-				// Disable clipping for Scrollviewer (edge seems to disable scrolling if 
-				// the clipping is enabled to the size of the scrollviewer, even if overflow-y is auto)
-				var clip = this is Controls.ScrollViewer ? (Rect?)null : new Rect(0, 0, newRect.Width, newRect.Height);
+				Rect? getClip()
```

Comment:
Why do you create a delegate for that ?

Reply:
... more than a delegate: a _display class_ capturing `this`.

Reply:
It's not a delegate, it's a local function. But it's that or a triple-if with a local variable... I want C# expression pattern matching :(

Reply:
Yes you're right it's a local function ... which will be translated as a display class by the compiler, like if you have declared a delegate 🤓

When do we update to C# 8 ?

Reply:
Nope, it's not a like a delegate, there's no indirection, there's a display class, but that's it.

I wrote something like this originally:

```
var clip = this is Controls.ScrollViewer ? (Rect?)null : Clip != null ? Clip.Rect : new Rect(0, 0, newRect.Width, newRect.Height);
```

because having to write `Rect clip;` then a bunch of if else, is not particularly nice either...

Reply:
👌